### PR TITLE
changed stream type & added push(null) to end stream

### PR DESCRIPTION
**Pull request by bergos**

### CODE REVIEW

File: lib/sequence.js [hunk 1]
```diff
@@ -13,15 +13,15 @@ class Sequence extends Readable {
...
-        pipeline.read()
-        return promise.then(next)
-      })
-    }, Promise.resolve())
-  };
+  async runPipelines () {
+    for (const pipeline of this.pipelines) {
+      pipeline.resume()
+
+      await promisify(finished)(pipeline)
```
Comment (tpluscode) [inline lib/sequence.js]: What is the `finished` import?

File: lib/sequence.js [hunk 2]
```diff
@@ -13,15 +13,15 @@ class Sequence extends Readable {
...
-      return promise.then(() => {
-        const next = eventToPromise(pipeline, 'end')
-        pipeline.read()
-        return promise.then(next)
-      })
-    }, Promise.resolve())
-  };
+  async runPipelines () {
+    for (const pipeline of this.pipelines) {
+      pipeline.resume()
```
Comment (tpluscode) [inline lib/sequence.js]: Is that method of built-in stream API?

File: lib/sequence.js (hunk 1 above)
Comment (bergos) [reply lib/sequence.js]: It was recently added to the stream package. It's a callback version of the initial `run` function in the core package (now already replaced with `finished`). It waits for the end of a stream and also handles errors. Now, that almost everything in a pipeline is a stream, I thought it's better to use standard stream functions for consistency.

File: lib/sequence.js (hunk 2 above)
Comment (bergos) [reply lib/sequence.js]: Yes, [Readable.resume](https://nodejs.org/api/stream.html#stream_readable_resume), it also has the benefit of consuming all chunks and not just a single one. Was actually not a problem with the pipeline like it is now, but when I was testing and writing chunks in the sub pipelines, it didn't reach the end. Btw. the base package should get an [operation](https://github.com/zazuko/barnard59-main/issues/16) `concat`, which could also handle this case.

**Pull request merged**